### PR TITLE
IBC Rate Limits: RBAC Permissions 2

### PR DESCRIPTION
**Pull request by nicolaslara**
## What is the purpose of the change

This is the same as https://github.com/osmosis-labs/osmosis/pull/8481 but with the correct bytecode to get tests to pass and changes to get lints to pass

## Testing and Verifying
all tests should pass

## Documentation and Release Note

  - [ ] Does this pull request introduce a new feature or user-facing behavior changes?
  - [ ] Changelog entry added to `Unreleased` section of `CHANGELOG.md`?

Where is the change documented?
  - [ ] Specification (`x/{module}/README.md`)
  - [ ] Osmosis documentation site
  - [ ] Code comments?
  - [ ] N/A

### CODE REVIEW

File: x/ibc-rate-limit/contracts/rate-limiter/src/state/flow.rs [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+    pub fn is_expired(&self, now: Timestamp) -> bool {
+        self.period_end < now
+    }
+
+    // Mutating methods
+
+    /// Expire resets the Flow to start tracking the value transfer from the
+    /// moment this method is called.
+    pub fn expire(&mut self, now: Timestamp, duration: u64) {
+        self.inflow = Uint256::from(0_u32);
```
Comment (crnbarr93) [inline x/ibc-rate-limit/contracts/rate-limiter/src/state/flow.rs:90]: Nit: Super small but using `Uint256::zero()` is a bit cleaner. Same for line below.

File: x/ibc-rate-limit/contracts/rate-limiter/src/state/rate_limit.rs [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+// transaction started executing
+fn calculate_channel_value(
+    channel_value: Uint256,
+    denom: &str,
+    funds: Uint256,
+    direction: &FlowType,
+) -> Uint256 {
+    match direction {
+        FlowType::Out => {
+            if denom.contains("ibc") {
```
Comment (crnbarr93) [inline x/ibc-rate-limit/contracts/rate-limiter/src/state/rate_limit.rs:88]: Nit: Any reason to not use `starts_with`?

Comment (nicolaslara) [reply x/ibc-rate-limit/contracts/rate-limiter/src/state/rate_limit.rs:88]: no reason. We can change it. 
```suggestion
            if denom.starts_with("ibc") {
```

File: x/ibc-rate-limit/contracts/rate-limiter/src/state/flow.rs (hunk 1 above)
Comment (nicolaslara) [reply x/ibc-rate-limit/contracts/rate-limiter/src/state/flow.rs:90]:
```suggestion
        self.inflow = Uint256::zero()
```
Sure! sounds good!

**Comment by nicolaslara** [comment]
(will have to recompile the contracts if we commit those suggestions)

**Review by crnbarr93 — APPROVED** [review]
LGTM! Both suggestions were super small nits and can easily be ignored for a future update if desired.

**Review by PaddyMc — APPROVED** [review]
:zap:

**Pull request merged**